cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -167,6 +167,7 @@
 
   private Job job;
   private Runnable cleanupTask;
+  private int metricsEmitIntervalInSecs;
```

Comment:
Why this has to be a field? I think you can turn the whole logic into a private method `getMetricsReportIntervalMillis(CConfiguration cConf, Map<String, String> runtimeArgs)`